### PR TITLE
fix(storage): new bloom filter that bind index with Column Id instead of column name

### LINKED ISSUE #10016: bug:  bloom filter should bind index with Column Id instead of column name
**Summary**

currently, the bloom index is bound with the column name of the data being indexed:

https://github.com/datafuselabs/databend/blob/0d816fc0d60ad807f2e4e7ee994a501bb7406566/src/query/storages/common/index/src/bloom_index.rs#L325-L329

This is NOT compatible with the way of table schema evaluation, it should be bound with the column id.

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Use column Id instead of column name to bind bloom filter index.

Closes #10016

### CODE REVIEW

File: src/query/expression/tests/it/schema.rs [hunk 1]
```diff
@@ -166,8 +167,7 @@ fn test_schema_from_struct() -> Result<()> {
     let schema = create_test_complex_schema();
     let flat_column_ids = schema.to_leaf_column_ids();
 
-    let (leaf_column_ids, leaf_fields) = schema.leaf_fields();
-    assert_eq!(leaf_column_ids, schema.to_leaf_column_ids());
```
Comment (lichuang) [inline src/query/expression/tests/it/schema.rs:170]: if remove `assert_eq!(leaf_column_ids, schema.to_leaf_column_ids());`, MUST add other test that merge leaf_fields.column_ids() to a vector equal to `schema.to_leaf_column_ids()`

File: src/query/expression/src/schema.rs [hunk 2]
```diff
@@ -594,32 +594,49 @@ impl TableSchema {
     }
 
     // return leaf fields and column ids
```
Comment (lichuang) [inline src/query/expression/src/schema.rs:596]: `return leaf fields and column ids` to `return leaf fields with column id`

File: src/query/storages/fuse/src/pruning/bloom_pruner.rs [hunk 3]
```diff
@@ -114,10 +123,25 @@ impl BloomPrunerCreator {
...
+        &self,
+        index_location: &Location,
+        index_length: u64,
+        column_ids: Vec<ColumnId>,
+    ) -> Result<bool> {
+        let version = index_location.1;
+        let index_columns = self.index_fields.iter().try_fold(
+            Vec::with_capacity(self.index_fields.len()),
+            |mut acc, field| {
+                if column_ids.contains(&field.column_id()) {
```
Comment (dantengsky) [inline src/query/storages/fuse/src/pruning/bloom_pruner.rs:136]: :+1: :+1: ,   mind adding some comments for it?

**Review by dantengsky — APPROVED** [review]

**Comment by dantengsky** [comment]
@lichuang  test of `leaf_fields.column_ids() to a vector equal to schema.to_leaf_column_ids()` added, PTAL

**Review by lichuang — APPROVED** [review]

**Pull request merged**